### PR TITLE
Added in "random" spectra that can be used for tests

### PR DESCRIPTION
**Pull request by brechmos-stsci**
I added in a fixture that defines several "random" spectra that could be used for the arithmetic, smoothing, fitting and any other tests.  And added an example in the arithmetic test to add two spectra.

Obviously, we can add any other spectra we want in order to confirm things such as adding two spectra of different units is helpful etc.  

Consider this the framework...

### CODE REVIEW

**Comment by brechmos-stsci** [comment]
I can modify the base flux spectrum if that is desired. It currently looks like this:
image

**Comment by brechmos-stsci** [comment]
This type of framework is what @SaOgaz could base other arithmetic tests on...

**Comment by brechmos-stsci** [comment]
@stscicrawford @nmearl If you guys are good with this PR, @SaOgaz would like to start using it for some tests etc.  Could one of you guys merge (or get back to me if you want changes!)?  (I can do the merge, if you want, but I don't feel comfortable without someone saying "go for it").

**Comment by nmearl** [comment]
I find this to be a bit confusing upon looking at it initially, I had to read the code closely to get what you were doing. My fear is that users in the future aren't going to understand the purpose of this -- in fact, I'm not quite sure what this gains us over the current implementation? Having a single source of pre-constructed spectrum objects? In that case I feel like having the fixtures implemented in their respective test files as unbound functions would make this more readable, and would avoid constructing some monolithic container class whose abstraction is, initially, a bit hard to parse. But honestly, I don't have a strong opinion. Perhaps @eteq could weigh in.

**Comment by brechmos-stsci** [comment]
The idea is to minimize the re-creation of example spectra for each of the test scripts (test_arithmetic.py, test_smoothing.py, test_linefitting.py) etc.

**Comment by stscicrawford** [comment]
Something like this is helpful and we've used something like this for testing NDData and CCDData.   Here is the example from ccdproc:
https://github.com/astropy/ccdproc/blob/888fb27f9cc43a4fb6a0c79c5fa9a2056dc7767a/ccdproc/tests/pytest_fixtures.py#L41

It would be useful to be able to set some scaling like
- number of peaks
- level of noise
-  number of elements

This also needs more documentation to help people with how it would work.

**Comment by brechmos-stsci** [comment]
Thanks @stscicrawford that is helpful, I'll take a look at it this afternoon.

**Comment by eteq** [comment]
(sorry that that might have been mixed messages: I was approving the idea of using fixtures, but may or may not be OK with the detail of it, depending on the responses to my inline comment)

File: specutils/tests/spectral_examples.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    def s1_AA_mJy_e1(self):
+        return self._s1_AA_mJy_e1
+
+    @property
+    def s1_AA_mJy_e1_flux(self):
+        return self._flux_e1
+
+
+@pytest.fixture
+def define_spectra():
```
[the remark this answers is not in the document]
Comment (brechmos-stsci) [reply specutils/tests/spectral_examples.py]: The file is `spectral_examples.py` and the test fixture is now called `simulated_spectra`.  I did this, in part with discussions with @crawfordsm, in the future it might be helpful to add a few *real* example spectra.

File: specutils/tests/spectral_examples.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import astropy.units as u
+from astropy.modeling import models
+from specutils.spectra import Spectrum1D
+import pytest
+
+class SpectraExamples(object):
```
[the remark this answers is not in the document]
Comment (brechmos-stsci) [reply specutils/tests/spectral_examples.py]: The summary of our discussion offline. This is a fixture container class that has several spectra defined in it.  This makes it easier to pass in the wavelengths, flux and resulting Spectrum1D objects for easier testing.  I added documentation regarding this since our discussion.

**Comment by brechmos-stsci** [comment]
Update....

@stscicrawford That is a great idea to do the parameterization of example simulated spectral creation, but I would suggest that could be for a round two of this test fixture. (more documentation added as well, along with an example in test_arithmetic.py).

@eteq We talked and I added much more documentation in the test fixture for clarity.  I also added an example in the test fixture file, and, as well, added one a new arithmetic test using the example simulated spectra.

**Comment by eteq** [comment]
Alright looks good.  @brechmos-stsci, I added a bit more clarification from our discussion yesterday to explain why this is a distinct class versus using a `Spectrum1D` directly.  So should be all good now.
